scripts/sign_image.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash -ex
+
+
+IMAGE=""
+SIGNING_KEY=""
+SIGNING_CERT=""
+CA_CERT=""
+
+usage()
+{
+    echo "Usage:  $0 -i <image_path> [-k <signing_key> -c <signing_cert> -a <ca_cert>]"
+    exit 1
+}
+
+generate_signing_key()
+{
+    TMP_CERT_PATH=$(mktemp -d)
+    SIGNING_KEY="${TMP_CERT_PATH}/signing.key"
+    SIGNING_CERT="${TMP_CERT_PATH}/signing.crt"
+    SIGNING_CSR="${TMP_CERT_PATH}/signing.csr"
+    CA_KEY="${TMP_CERT_PATH}/ca.key"
+
+    # Generate the CA key and certificate
+    openssl genrsa -out $CA_KEY 4096
+    openssl req -x509 -new -nodes -key $CA_KEY -sha256 -days 3650 -subj "/C=Test/ST=Test/L=Test/O=Test/CN=Test" -out $CA_CERT
+
+    # Generate the signing key, certificate request and certificate
+    openssl genrsa -out $SIGNING_KEY 4096
+    openssl req -new -key $SIGNING_KEY -subj "/C=Test/ST=Test/L=Test/O=Test/CN=Test" -out $SIGNING_CSR
+    openssl x509 -req -in $SIGNING_CSR -CA $CA_CERT -CAkey $CA_KEY -CAcreateserial -out $SIGNING_CERT -days 1825 -sha256
+}
+
+while getopts "i:k:c:a:t:" opt; do
+    case $opt in
+        i)
+            IMAGE=$OPTARG
+            ;;
+        k)
+            SIGNING_KEY=$OPTARG
+            ;;
+        c)
+            SIGNING_CERT=$OPTARG
+            ;;
+        a)
+            CA_CERT=$OPTARG
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+
+[ -z $CA_CERT ] && echo "Not to sign the image since the CA certificate not provided" && exit 0
```

Comment:
>Not to sign the image since the CA certificate not provide [](start = 25, length = 58)

Not to sign the image since the CA certificate path is not provided

Reply:
I mean to change the error message

---
In reply to: [429657716](https://github.com/Azure/sonic-buildimage/pull/4627#discussion_r429657716) [](ancestors = 429657716)

Reply:
Fixed.